Jorgen master

CarlosDerSeher opened the pull request:
Integrated syncing solution into master as good as I could. I didn't test all the functionality, e.g. OTA but seems to work quite stable even with DSP applied to audio data

Best Regards
Karl

CarlosDerSeher committed:
-) add component median filter
-) add player module to lightsnapcast component which can be used to do synced playback on Boards with PSRAM
-) in snapcast module of lightsnapcast component use PSRAM for wirechunk payload if available
-) external opus component throws an error on compilation, use -Wno-maybe-uninitialized on file opus/silk/quant_LTP_gains.c to prevent this
-) integrate player module in main.c
   o dropped dsp_processor component
     TODO: do signal processing right after decoding on wire chunk data
           and put them in the player queue so there won't be any latency
           introduced from DSP
   o use esp timer to send snapcast time sync messages
   o use send/recv for socket operations also do error checking after any of the socket API calls
-) change a lot of settings in sdkconfig
   o mainly WiFi and LwIP related buffer size increasing
   o increased freeRTOS tick rate from 100Hz t0 1000z
   o increase SPI RAM clock from 40MHz to 80MHz
   o enable SPIRAM_2T_MODE because of failing boot if it is not

CarlosDerSeher committed:
-) adapt dsp_processor component so it can be used with player of snapcast component
-) adapt ota_server component so it can be used with player of snapcast component
-) use dsp_processor() on decoded chunk data in main()
-) add deinit function to player module in snapcast component

CarlosDerSeher committed:
-) add wifi profisioning to wifi_interface
   o can be enabled through menuconfig
-) use custom partitions.csv because default does not fit app

jorgenkraghjakobsen commented:
Hi Karl  - That is quit some changes you have added. Nice I will dive in and check if i can get it to play/sync here.

CarlosDerSeher commented:
I have to add that it will only work on boards with spi ram attached. Not sure if it will even be possible to do good syncing on internal ram only. Maybe with a cable connection...

CarlosDerSeher committed:
- minimize RAM usage of all components
- use both IRAM and DRAM in player component soo we can buffer up to 1s on modules without SPI RAM
- support fragemented pcm chunks so we can use all available RAM if there isn't a big enough block available but still enough HEAP
  o add player insert functions which handles this stuff
  o start to remove static variable declarations from dsp_processor component
    and also reuse buffers as much as possible
- add PCM data configuration to dsp_processor KCONFIG

jorgenkraghjakobsen commented:
Hi Karl 
I am do a lot of change on my side as well and the sync structure you are suggesting is very different from the way I am doing it 
I still work on a component style snap client. But still a lot to do.
